catalogs/catalog-hive/src/main/java/com/datastrato/gravitino/catalog/hive/CachedClientPool.java

```diff
@@ -65,7 +65,14 @@ public class CachedClientPool implements ClientPool<IMetaStoreClient, TException
     this.clientPoolCache =
         Caffeine.newBuilder()
             .expireAfterAccess(evictionInterval, TimeUnit.MILLISECONDS)
-            .removalListener((ignored, value, cause) -> ((HiveClientPool) value).close())
+            .removalListener(
+                (ignored, value, cause) -> {
+                  // In fact, we have closed some HiveClientPools manually in the `close` method.
+                  // to avoid closing it repeatedly, we add a judgment that if it has been closed,
```

Comment:
Where is the "judgment" you mentioned here?

 I'm super confused about the comment here. You write lots of words here, but doesn't reflect the code here. If it is not here, please add the comment to the code where it happens.